### PR TITLE
Add Documentation and Configuration Files

### PR DESCRIPTION
**Pull request by ytupman**
This PR adds essential documentation and configuration files to the repository, including:

- CODEOWNERS
- CODE_OF_CONDUCT.md
- CONTRIBUTING.md
- README.md enhancements
- SECURITY.md
- MkDocs configuration and documentation files

These changes enhance project documentation, establish contribution and security guidelines, and configure MkDocs for documentation generation.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+# Welcome to aws_resource_validator
```
Comment (CoreOxide) [inline docs/index.md:2]: what's this file?
it has the same text as in the `README`

File: docs/usage.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+# Usage
```
Comment (CoreOxide) [inline docs/usage.md:2]: Also, this is part of the `README` as well.

File: mkdocs.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+site_name: aws_resource_validator
```
Comment (CoreOxide) [inline mkdocs.yml:1]: what does this file do?

File: mkdocs.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+theme:
+  name: material
+
+nav:
+  - Home: index.md
+  - Usage: usage.md
+  - API Reference: api.md
+
+plugins:
+  - search
```
Comment (CoreOxide) [inline mkdocs.yml:11]: what does this plugin mean?

File: docs/index.md (hunk 1 above)
Comment (ytupman) [reply docs/index.md:2]: It's for GitHub Pages

File: docs/usage.md (hunk 2 above)
Comment (ytupman) [reply docs/usage.md:2]: It's for GitHub Pages

File: mkdocs.yml (hunk 3 above)
Comment (ytupman) [reply mkdocs.yml:1]: The mkdocs.yml file is a configuration file for MkDocs, which is used for GitHub Pages.

File: mkdocs.yml (hunk 4 above)
Comment (ytupman) [reply mkdocs.yml:11]: This plugin enables a full-text search feature on the documentation site, allowing users to easily search for keywords or phrases within your site's content.

**Review by CoreOxide — APPROVED** [review]

**Pull request merged**